server/internal/docker/executor.go

```diff
@@ -0,0 +1,157 @@
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/mount"
+)
+
+// pullImage pulls the Docker image if needed
+func (r *Runner) pullImage(ctx context.Context, imageName string, version string) error {
+	// Only pull if version is "latest" or image doesn't exist locally
+	if version == "latest" {
+		logs.Info("Pulling Docker image: %s", imageName)
+
+		reader, err := r.dockerClient.ImagePull(ctx, imageName, image.PullOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to pull image %s: %v", imageName, err)
+		}
+		defer reader.Close()
+
+		// Read the pull output (optional, for logging)
+		_, err = io.Copy(io.Discard, reader)
+		if err != nil {
+			logs.Warning("Failed to read image pull output: %v", err)
+		}
+	}
+	return nil
+}
+
+// ExecuteDockerCommand executes a Docker command with the given parameters using Docker SDK
+func (r *Runner) ExecuteDockerCommand(ctx context.Context, flag string, command Command, sourceType, version, configPath string, additionalArgs ...string) ([]byte, error) {
+	outputDir := filepath.Dir(configPath)
+	if err := utils.CreateDirectory(outputDir, DefaultDirPermissions); err != nil {
+		return nil, err
+	}
+
+	imageName := r.GetDockerImageName(sourceType, version)
+
+	// Pull image if necessary
+	if err := r.pullImage(ctx, imageName, version); err != nil {
+		return nil, err
+	}
+
+	// Build command arguments
+	cmdArgs := r.buildCommandArgs(flag, command, configPath, additionalArgs...)
+
+	// Get host output directory for volume mounting
+	hostOutputDir := r.getHostOutputDir(outputDir)
+
+	// Create container configuration
+	containerConfig := &container.Config{
+		Image: imageName,
+		Cmd:   cmdArgs,
+	}
+
+	// Create host configuration with volume mounts
+	hostConfig := &container.HostConfig{
+		Mounts: []mount.Mount{
+			{
+				Type:   mount.TypeBind,
+				Source: hostOutputDir,
+				Target: ContainerMountDir,
+			},
+		},
+		AutoRemove: true, // Automatically remove container when it exits
+	}
+
+	logs.Info("Running Docker container with image: %s, command: %v", imageName, cmdArgs)
+
+	// Create container
+	resp, err := r.dockerClient.ContainerCreate(ctx, containerConfig, hostConfig, nil, nil, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create container: %v", err)
+	}
+
+	// Start container
+	if err := r.dockerClient.ContainerStart(ctx, resp.ID, container.StartOptions{}); err != nil {
+		return nil, fmt.Errorf("failed to start container: %v", err)
+	}
+
+	// Wait for container to finish
+	statusCh, errCh := r.dockerClient.ContainerWait(ctx, resp.ID, container.WaitConditionNotRunning)
+	select {
+	case err := <-errCh:
+		if err != nil {
+			return nil, fmt.Errorf("error waiting for container: %v", err)
+		}
+	case status := <-statusCh:
+		if status.StatusCode != 0 {
+			// Get container logs for error details
+			logs, _ := r.getContainerLogs(ctx, resp.ID)
+			return nil, fmt.Errorf("container exited with status %d: %s", status.StatusCode, logs)
```

Comment:
here `logs` is []byte so we may need to stringify it when logging. 
```suggestion
			return nil, fmt.Errorf("container exited with status %d: %s", status.StatusCode, string(logs))
```